src/libraries/System.Private.CoreLib/src/System/IndexOfAnyAsciiSearcher.cs

```diff
@@ -0,0 +1,345 @@
+            // - Values  > 32767 result in 0. Because of this we must do more work to handle needles that contain 0.
+            // ARM64: Take the low byte of each character.
+            // - All values result in (value & 0xFF).
+            Vector128<byte> source = Sse2.IsSupported
+                ? Sse2.PackUnsignedSaturate(source0, source1)
+                : AdvSimd.Arm64.UnzipEven(source0.AsByte(), source1.AsByte());
+
+            // On X86, the Ssse3.Shuffle instruction will already perform an implicit 'AND 0xF' on the indices, so we can skip it.
+            // For values above 127, Ssse3.Shuffle will also set the result to 0. This saves us from explicitly checking whether the input was ascii below.
+            Vector128<byte> lowerNibbles = Ssse3.IsSupported
+                ? source
+                : source & Vector128.Create((byte)0xF);
+
+            Vector128<byte> highNibbles = Vector128.ShiftRightLogical(source.AsInt32(), 4).AsByte() & Vector128.Create((byte)0xF);
+
+            // The bitmapLookup represents a 8x16 table of bits, indicating whether a character is present in the needle.
+            // Lookup the rows via the lower nibble and the column via the higher nibble.
+            Vector128<byte> bitMask = Shuffle(bitmapLookup, lowerNibbles);
+            Vector128<byte> bitPositions = Shuffle(Vector128.Create(0x8040201008040201).AsByte(), highNibbles);
+
+            Vector128<byte> result = bitMask & bitPositions;
+
+            // On ARM64, we ignored the high byte of every character when packing (see above).
+            // The 'result' can therefore contain false positives - e.g. 0x141 would match 0x41 ('A').
+            // On X86, PackUnsignedSaturate resulted in values becoming 0 for inputs above 32767.
+            // Any value above 32767 would therefore match against 0. If 0 is present in the needle, we must clear the false positives.
+            // In both cases, we can correct the result by clearing any bits that matched with a non-ascii source character.
+            if (AdvSimd.Arm64.IsSupported || TOptimizations.NeedleContainsZero)
+            {
+                Vector128<short> ascii0 = Vector128.LessThan(source0.AsUInt16(), Vector128.Create((ushort)128)).AsInt16();
+                Vector128<short> ascii1 = Vector128.LessThan(source1.AsUInt16(), Vector128.Create((ushort)128)).AsInt16();
+                Vector128<byte> ascii = Sse2.IsSupported
+                    ? Sse2.PackSignedSaturate(ascii0, ascii1).AsByte()
+                    : AdvSimd.Arm64.UnzipEven(ascii0.AsByte(), ascii1.AsByte());
+                result &= ascii;
+            }
+
+            return TNegator.NegateIfNeeded(result);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector128<byte> Shuffle(Vector128<byte> vector, Vector128<byte> indices)
+        {
+            // Vector128.Shuffle is emitting a software fallback as indices are not given as inline constants.
```

Comment:
Is there an issue for this? If Sse3.Shuffle can do the "right thing", presumably Vector128.Shuffle should be able to do the right thing if Ssse33.IsSupported.

Reply:
This one is about Vector256, but I believe it's the same problem: https://github.com/dotnet/runtime/issues/72793

@tannergooding `Vector128.Shuffle` may emit extra code here to match the behavior across different platforms (not just `Ssse3.Shuffle` and `AdvSimd.Arm64.VectorTableLookup` directly), right?
If that's the case, I can remove the comment here as we wouldn't actually want to use `Vector128.Shuffle`, as we don't need exactly the same behavior.